src/Internal/Line.elm

```diff
@@ -29,14 +32,51 @@ view { toSvgCoords } { style, customAttrs } points =
             startPath svgPoints
 
         instructions =
-            coordsToInstruction "L" svgPoints
+            case smoothing of
+                None ->
+                    coordsToInstruction "L" svgPoints
+                Cosmetic ->
+                    coordsListToInstruction "S" (coordsToSmoothBezierCoords svgPoints)
 
         attrs =
             (stdAttributes (startInstruction ++ instructions) style) ++ customAttrs
     in
         Svg.path attrs []
 
 
+coordsToSmoothBezierCoords : List Point -> List ( List ( Point ) )
+coordsToSmoothBezierCoords coords =
+    let
+        first = List.head coords |> Maybe.withDefault (0, 0)
+        last = List.reverse coords |> List.head |> Maybe.withDefault (0, 0)
+        paddedCoords = [first] ++ coords ++ [last]
```

Comment:
Hey @mathiasbaert! It all looks really great! I have a question though: Why do you add the `first` padding here? In the example it causes the line to go outside the coordinates given by the user, but only on the left side of the path. Is that on purpose? P.S. Thanks for fixing the type sharing (reg. `Plot.Types`)!